### ISSUE TITLE
SignUp return both Session and User

### ISSUE THREAD

**Issue by fplaras** [supabase-community/supabase-csharp#18]
There is a use case in the documentation that indicates either object is returned based on email confirmation requirement but the current signup() only returns the Session object. 

Ref: [https://supabase.com/docs/reference/javascript/auth-signup](https://supabase.com/docs/reference/javascript/auth-signup)
```
const { user, session, error } = await supabase.auth.signUp({
  email: 'example@email.com',
  password: 'example-password',
})
```

I am thinking it could return a Tuple that includes the User and Session objects.

**Comment by acupofjose** [comment]
@fplaras [A `Session` includes the `User` as a property](https://supabase-community.github.io/supabase-csharp/api/Supabase.Gotrue.Session.html#Supabase_Gotrue_Session_User) - is that different than what you're asking?

**Comment by fplaras** [comment]
@acupofjose yes, because when a user is signed up and their email is not confirmed)with requirement) session is null but if I could retrive the user data that was created I can insert for example user profile records that reference the new user uuid but if session is null I would have to wait for the email confirmation to go through to then retrive the user based on the access token to then create the profile and insert profile records.

**Comment by acupofjose** [comment]
Ah! I see - I'll look into it, let me make an issue on the Gotrue page

**Comment by acupofjose** [comment]
@fplaras `Session` will now be returned with `null` token data but a populated User property. Available in `gotrue-csharp@2.3.3` and `supabase-csharp@0.2.12`.

**Comment by fplaras** [comment]
> @fplaras `Session` will now be returned with `null` token data but a populated `User` property. Available in `gotrue-csharp@2.3.3` and `supabase-csharp@0.2.12`.

Sweet! Thank you
